packages/rx_bloc_cli/mason_templates/bricks/lib_auth_apple/__brick__/lib/lib_auth_apple/ui_components/apple_login_button.dart

```diff
@@ -0,0 +1,24 @@
+{{> licence.dart }}
+
+import 'package:flutter/material.dart';
+import 'package:flutter_rx_bloc/flutter_rx_bloc.dart';
+import 'package:sign_in_with_apple/sign_in_with_apple.dart';
+
+import '../blocs/apple_login_bloc.dart';
+
+/// [AppleLoginButton] provides out of the box Log in with Apple
+/// functionality along with default view of the button.
+///
+/// In order to make it work, provide LoginWithAppleBlocType to the context
+/// above this widget or use [AppleLoginButtonWithDependencies].
+class AppleLoginButton extends StatelessWidget {
+  const AppleLoginButton({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) => RxBlocBuilder<AppleLoginBlocType, bool>(
```

Comment:
Do we really need this widget ?

Reply:
Currently we just wrap the provided from the package button with builder to disable it, when the state is loading. All that code might be moved to AppleLoginButtonWithDependencies.

Reply:
Sounds good to me